compiler/src/dotty/tools/dotc/transform/UnusedChecks.scala

```diff
@@ -0,0 +1,102 @@
+package dotty.tools.dotc.transform
+
+import dotty.tools.dotc.ast.Trees._
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.core.Contexts._
+import dotty.tools.dotc.core.Decorators._
+import dotty.tools.dotc.core.Flags._
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.transform.MegaPhase.MiniPhase
+
+/** This phase checks that unused `def`s, `val`s and parameters are only used in valid place.
+ *  It also checks that the `unused` keyword is only used on `def`, `val` and parameters lists
+ *
+ *  Unused values can be used:
+ *    - As parameters to an unused function
+ *    - In statement position
+ *    - Anywhere in the RHS of an `unused def` or `unused val`
+ */
+class UnusedChecks extends MiniPhase {
+  import tpd._
+
+  override def phaseName: String = "unusedChecks"
+
+
+  /* Tree transform */
+
+  override def transformTypeDef(tree: TypeDef)(implicit ctx: Context): tree.type = {
+    if (tree.symbol.is(UnusedCommon))
```

Comment:
`is(Unused)`. `UnusedCommon` should never be used in an `is` check.

Reply:
Done